sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client.c

```diff
@@ -12,9 +12,153 @@
 
 #include <cmocka.h>
 
-int test_iot_provisioning_client()
+static const az_span test_global_device_endpoint
+    = AZ_SPAN_LITERAL_FROM_STR("global.azure-devices-provisioning.net");
+
+#define TEST_ID_SCOPE "0neFEEDC0DE"
+static const az_span test_id_scope = AZ_SPAN_LITERAL_FROM_STR(TEST_ID_SCOPE);
+
+#define TEST_REGISTRATION_ID "myRegistrationId"
+static const az_span test_registration_id = AZ_SPAN_LITERAL_FROM_STR(TEST_REGISTRATION_ID);
+
+#define TEST_USER_AGENT "c/1.2.3"
+#define TEST_REQUEST_ID "9060edd6-cc37-43d7-a96e-1bb3c3abcb82"
+
+static void test_az_iot_provisioning_client_options_default_succeed()
+{
+  az_iot_provisioning_client_options options = az_iot_provisioning_client_options_default();
+  assert_true(az_span_is_content_equal(options.user_agent, AZ_SPAN_NULL));
+}
+
+static void test_az_iot_provisioning_client_default_options_get_connect_info_succeed()
+{
+  az_iot_provisioning_client client;
+  az_result ret = az_iot_provisioning_client_init(
+      &client, test_global_device_endpoint, test_id_scope, test_registration_id, NULL);
+  assert_int_equal(AZ_OK, ret);
+
+  uint8_t client_id_buffer[128];
+  az_span client_id = AZ_SPAN_LITERAL_FROM_BUFFER(client_id_buffer);
```

Comment:
@gilbertw @barcharcraz @JeffreyRichter this is actually not accepted by L4 warnings on the Microsoft Compiler:

```
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(41,1): error C2220: the following warning is treated as an error [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(41,1): warning C4221: nonstandard extension used: 'ptr': cannot be initialized using address of automatic variable 'client_id_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(40): message : see declaration of 'client_id_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(41,1): warning C4204: nonstandard extension used: non-constant aggregate initializer [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(51,1): warning C4221: nonstandard extension used: 'ptr': cannot be initialized using address of automatic variable 'username_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(50): message : see declaration of 'username_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(51,1): warning C4204: nonstandard extension used: non-constant aggregate initializer [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(76,1): warning C4221: nonstandard extension used: 'ptr': cannot be initialized using address of automatic variable 'username_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(75): message : see declaration of 'username_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(76,1): warning C4204: nonstandard extension used: non-constant aggregate initializer [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(94,1): warning C4221: nonstandard extension used: 'ptr': cannot be initialized using address of automatic variable 'topic_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(93): message : see declaration of 'topic_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(94,1): warning C4204: nonstandard extension used: non-constant aggregate initializer [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(112,1): warning C4221: nonstandard extension used: 'ptr': cannot be initialized using address of automatic variable 'topic_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(111): message : see declaration of 'topic_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(112,1): warning C4204: nonstandard extension used: non-constant aggregate initializer [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(129,1): warning C4221: nonstandard extension used: 'ptr': cannot be initialized using address of automatic variable 'topic_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(128): message : see declaration of 'topic_buffer' [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
D:\a\1\s\sdk\iot\provisioning\tests\cmocka\test_az_iot_provisioning_client.c(129,1): warning C4204: nonstandard extension used: non-constant aggregate initializer [D:\a\1\s\build\sdk\iot\provisioning\tests\cmocka\az_iot_provisioning_test.vcxproj]
```

Reply:
yea this is why all the tests I have for other parts as `az_span_init()` :/

Reply:
Looks to me like a bug in the macro that needs to be fixed. If this is fine and it'll work, we need to add the correct pragmas to disable the warnings.

Reply:
This is indeed a bug but in msvc. I'm switching to az_span_init.